src/core/pyopenscad.cc

```diff
@@ -0,0 +1,261 @@
+
+    {"projection", (PyCFunction) python_projection, METH_VARARGS | METH_KEYWORDS, "Projection Object."},
+    {"surface", (PyCFunction) python_surface, METH_VARARGS | METH_KEYWORDS, "Surface Object."},
+    {"osimport", (PyCFunction) python_import, METH_VARARGS | METH_KEYWORDS, "Import Object."},
+    {"color", (PyCFunction) python_color, METH_VARARGS | METH_KEYWORDS, "Import Object."},
+
+    {"output", (PyCFunction) python_output, METH_VARARGS | METH_KEYWORDS, "Output the result."},
+    {"version", (PyCFunction) python_version, METH_VARARGS | METH_KEYWORDS, "Output openscad Version."},
+    {"version_num", (PyCFunction) python_version_num, METH_VARARGS | METH_KEYWORDS, "Output openscad Version."},
+    {NULL, NULL, 0, NULL}
+};
+// TODO missing 
+//  dxf_dim 
+//  dxf_cross 
+//  intersection_for 
+//  dxf_linear_extrude 
+//  dxf_rotate_extrude 
+//  fontmetrics 
+
+PyObject *python_oo_args(PyObject *self, PyObject *args)
+{
+	int i;
+	PyObject *new_args = PyTuple_New(PyTuple_Size(args)+1);
+//	Py_INCREF(&new_args);
+	PyTuple_SetItem(new_args,0,self);
+	for(i=0;i<PyTuple_Size(args);i++)
+		PyTuple_SetItem(new_args,i+1,PyTuple_GetItem(args,i));
+	return new_args;
+}
+
+void get_fnas(double &fn, double &fa, double &fs) {
+  PyObject *mainModule = PyImport_AddModule("__main__");
+  if(mainModule == NULL) return;
+  PyObject *varFn = PyObject_GetAttrString(mainModule, "fn");
+  PyObject *varFa = PyObject_GetAttrString(mainModule, "fa");
+  PyObject *varFs = PyObject_GetAttrString(mainModule, "fs");
+  if(varFn != NULL)  fn = PyFloat_AsDouble(varFn);
+  if(varFa != NULL)  fa = PyFloat_AsDouble(varFa);
+  if(varFs != NULL)  fs = PyFloat_AsDouble(varFs);
+}
+
+
+
+static PyMethodDef PyOpenSCADMethods[] = {
+    {"translate", (PyCFunction) python_translate_oo, METH_VARARGS | METH_KEYWORDS, "Move  Object."},
+    {"rotate", (PyCFunction) python_rotate_oo, METH_VARARGS | METH_KEYWORDS, "Rotate Object."},
+    {"scale", (PyCFunction) python_scale_oo, METH_VARARGS | METH_KEYWORDS, "Scale Object."},
+    {"mirror", (PyCFunction) python_mirror_oo, METH_VARARGS | METH_KEYWORDS, "Mirror Object."},
+    {"multmatrix", (PyCFunction) python_multmatrix_oo, METH_VARARGS | METH_KEYWORDS, "Multmatrix Object."},
+    {"linear_extrude", (PyCFunction) python_linear_extrude_oo, METH_VARARGS | METH_KEYWORDS, "Linear_extrude Object."},
+    {"rotate_extrude", (PyCFunction) python_rotate_extrude_oo, METH_VARARGS | METH_KEYWORDS, "Rotate_extrude Object."},
+    {"offset", (PyCFunction) python_offset_oo, METH_VARARGS | METH_KEYWORDS, "Offset Object."},
+    {"roof", (PyCFunction) python_roof_oo, METH_VARARGS | METH_KEYWORDS, "Roof Object."},
+    {"output", (PyCFunction) python_output_oo, METH_VARARGS | METH_KEYWORDS, "Output the result."},
+    {"color", (PyCFunction) python_color_oo, METH_VARARGS | METH_KEYWORDS, "Output the result."},
+    {NULL, NULL, 0, NULL}
+};
+
+
+PyTypeObject PyOpenSCADType = {
```

Comment:
You may be able to benefit from the fact that C++ zero-initializes structs, and initialize only the non-zero members, smth. like this:

```

PyTypeObject PyOpenSCADType = {
  .ob_base = PyVarObject_HEAD_INIT(NULL, 0),
  .tp_name = "PyOpenSCAD",

  ...etc for all non-zero field members
};
```